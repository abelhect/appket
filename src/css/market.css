/* .app-col {
  width: inherit;
  height: inherit;
} */
