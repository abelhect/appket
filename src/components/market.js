import React from 'react';
import {Grid, Row, Col, Thumbnail} from 'react-bootstrap';
import {connect} from 'react-redux';
import '../css/market.css';
// import PropTypes from 'prop-types';
import '../images/shoppingcart_thumb.png';
import '../images/reminderpro_thumb.png';
import '../images/pomodorotimer_thumb.png';
import '../images/goalcoach_thumb.png';

export const Market = (props) => (
    <Grid>
      <p className="App-intro">
        <em>Welcome to Papoâ€™s App Marketplace,
        where you will find a library of web and mobile applications
        that will make your life easier.</em>
      </p>
      <Row className="app-row">
        {
          props.list.map((app, index) => {
            return(
                <Col className="app-col" xs={12} sm={6} md={3} key={index} >
                  <Thumbnail href="#" alt={app.name} key={index} src={app.thumbnail}/>
                </Col>
            )
          })
        }
      </Row>
    </Grid>

);

// Market.propTypes = {
//   marketappslist: PropTypes.array.isRequired
// }

function mapStateToProps(state){
  return{
    list: state.list
  };
}

export default connect(mapStateToProps)(Market);
