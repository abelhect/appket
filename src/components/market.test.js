import React from 'react';
import '../setupTests';
import {shallow} from 'enzyme';
import {Market} from './market';

describe('All tests for the Market Component', ()=> {
  const props = {list:
    [
      {
          "name": "name 1",
          "intro": "intro 1",
          "thumbnail": "thumbnail 1"
      },
      {
          "name": "name 2",
          "intro": "intro 2",
          "thumbnail": "thumbnail 2"
      }

    ]
  };
  const market = shallow(<Market {...props}/>);

  it('renders correctly', () =>{
    expect(market).toMatchSnapshot();
  });

  it('renders the `app-row` in new Grid element', () =>{
    expect(market.find('.app-row').length).toEqual(1);
  });

  it('renders the `app-col` in new Row element', () =>{
    expect(market.find('.app-col').length).toEqual(2);
  });
});
