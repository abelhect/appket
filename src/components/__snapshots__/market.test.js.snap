// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All tests for the Market Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    list={
        Array [
            Object {
              "intro": "intro 1",
              "name": "name 1",
              "thumbnail": "thumbnail 1",
            },
            Object {
              "intro": "intro 2",
              "name": "name 2",
              "thumbnail": "thumbnail 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <p
          className="App-intro"
>
          <em>
                    Welcome to Papo’s App Marketplace, where you will find a library of web and mobile applications that will make your life easier.
          </em>
</p>,
        <Row
          bsClass="row"
          className="app-row"
          componentClass="div"
>
          <Col
                    bsClass="col"
                    className="app-col"
                    componentClass="div"
                    md={3}
                    sm={6}
                    xs={12}
          >
                    <Thumbnail
                              alt="name 1"
                              bsClass="thumbnail"
                              href="#"
                              src="thumbnail 1"
                    />
          </Col>
          <Col
                    bsClass="col"
                    className="app-col"
                    componentClass="div"
                    md={3}
                    sm={6}
                    xs={12}
          >
                    <Thumbnail
                              alt="name 2"
                              bsClass="thumbnail"
                              href="#"
                              src="thumbnail 2"
                    />
          </Col>
</Row>,
      ],
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <em>
            Welcome to Papo’s App Marketplace, where you will find a library of web and mobile applications that will make your life easier.
</em>,
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Papo’s App Marketplace, where you will find a library of web and mobile applications that will make your life easier.",
          },
          "ref": null,
          "rendered": "Welcome to Papo’s App Marketplace, where you will find a library of web and mobile applications that will make your life easier.",
          "type": "em",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": Array [
            <Col
              bsClass="col"
              className="app-col"
              componentClass="div"
              md={3}
              sm={6}
              xs={12}
>
              <Thumbnail
                            alt="name 1"
                            bsClass="thumbnail"
                            href="#"
                            src="thumbnail 1"
              />
</Col>,
            <Col
              bsClass="col"
              className="app-col"
              componentClass="div"
              md={3}
              sm={6}
              xs={12}
>
              <Thumbnail
                            alt="name 2"
                            bsClass="thumbnail"
                            href="#"
                            src="thumbnail 2"
              />
</Col>,
          ],
          "className": "app-row",
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <Thumbnail
                alt="name 1"
                bsClass="thumbnail"
                href="#"
                src="thumbnail 1"
/>,
              "className": "app-col",
              "componentClass": "div",
              "md": 3,
              "sm": 6,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "alt": "name 1",
                "bsClass": "thumbnail",
                "href": "#",
                "src": "thumbnail 1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <Thumbnail
                alt="name 2"
                bsClass="thumbnail"
                href="#"
                src="thumbnail 2"
/>,
              "className": "app-col",
              "componentClass": "div",
              "md": 3,
              "sm": 6,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "alt": "name 2",
                "bsClass": "thumbnail",
                "href": "#",
                "src": "thumbnail 2",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <p
            className="App-intro"
>
            <em>
                        Welcome to Papo’s App Marketplace, where you will find a library of web and mobile applications that will make your life easier.
            </em>
</p>,
          <Row
            bsClass="row"
            className="app-row"
            componentClass="div"
>
            <Col
                        bsClass="col"
                        className="app-col"
                        componentClass="div"
                        md={3}
                        sm={6}
                        xs={12}
            >
                        <Thumbnail
                                    alt="name 1"
                                    bsClass="thumbnail"
                                    href="#"
                                    src="thumbnail 1"
                        />
            </Col>
            <Col
                        bsClass="col"
                        className="app-col"
                        componentClass="div"
                        md={3}
                        sm={6}
                        xs={12}
            >
                        <Thumbnail
                                    alt="name 2"
                                    bsClass="thumbnail"
                                    href="#"
                                    src="thumbnail 2"
                        />
            </Col>
</Row>,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <em>
              Welcome to Papo’s App Marketplace, where you will find a library of web and mobile applications that will make your life easier.
</em>,
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Papo’s App Marketplace, where you will find a library of web and mobile applications that will make your life easier.",
            },
            "ref": null,
            "rendered": "Welcome to Papo’s App Marketplace, where you will find a library of web and mobile applications that will make your life easier.",
            "type": "em",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": Array [
              <Col
                bsClass="col"
                className="app-col"
                componentClass="div"
                md={3}
                sm={6}
                xs={12}
>
                <Thumbnail
                                alt="name 1"
                                bsClass="thumbnail"
                                href="#"
                                src="thumbnail 1"
                />
</Col>,
              <Col
                bsClass="col"
                className="app-col"
                componentClass="div"
                md={3}
                sm={6}
                xs={12}
>
                <Thumbnail
                                alt="name 2"
                                bsClass="thumbnail"
                                href="#"
                                src="thumbnail 2"
                />
</Col>,
            ],
            "className": "app-row",
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <Thumbnail
                  alt="name 1"
                  bsClass="thumbnail"
                  href="#"
                  src="thumbnail 1"
/>,
                "className": "app-col",
                "componentClass": "div",
                "md": 3,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "alt": "name 1",
                  "bsClass": "thumbnail",
                  "href": "#",
                  "src": "thumbnail 1",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <Thumbnail
                  alt="name 2"
                  bsClass="thumbnail"
                  href="#"
                  src="thumbnail 2"
/>,
                "className": "app-col",
                "componentClass": "div",
                "md": 3,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "alt": "name 2",
                  "bsClass": "thumbnail",
                  "href": "#",
                  "src": "thumbnail 2",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
